filebeat/docs/running-on-kubernetes.asciidoc

```diff
@@ -271,3 +271,71 @@ _Kubernetes is not responsible for rotating logs, but rather a deployment tool s
 Different logrotation strategies can cause issues that might make Filebeat losing events or even duplicating events.
 Users can find more information about Filebeat's logrotation best practises at Filebeat's
 <<file-log-rotation,log rotation specific documentation>>
+
+[float]
+==== Kubernetes metadata
+
+In case filebeat is configured with kubernetes autodiscover provider (suggested configuration), kubernetes metadata
+are collected by the provider internal mechanism. It uses watchers which watch for pod resources
+in the cluster and collects all container IDs and relevant information(labels, annotations, namespace info, node info etc.).
+For each one of the unique container IDs, a unique filestream configuration block is generated by replacing the dynamic variables `${data.kubernetes.container.id}` with each container's ID.
+Then filestrean input, parses the log file of each configuration block knowing to which actual container it belongs to. 
+Before publishing the log event, it enriches it with all the collected information and metadata.
+
+[source,yaml]
+------------------------------------------------
+filebeat.autodiscover:
+     providers:
+       - type: kubernetes
+         node: ${NODE_NAME}
+         templates:
+          - condition.contains:
+              kubernetes.namespace: "kube-system"
+            config:
+              - type: filestream
+                id: kubernetes-container-logs-${data.kubernetes.pod.name}-${data.kubernetes.container.id}
+                parsers:
+                - container: ~
+                prospector:
+                  scanner:
+                    fingerprint.enabled: true
+                    symlinks: true
+                file_identity.fingerprint: ~
+                processors:
+                  - add_fields:
+                      target: filebeat
+                      fields:
+                        template: comet
+                  - add_fields:
+                      target: filebeat
+                      fields:
+                        template: glencore-logIngestPipelineSuffix
+                paths:
+                  - /var/log/containers/*-${data.kubernetes.container.id}.log
+------------------------------------------------
+
+In case filebeat is configured with filestream input, kubernetes metadata are collected using the `add_kubernetes_metadata` processor. The processor, using matchers, tries to associate a log file detected by filestream input with an actual container.
```

Comment:
```suggestion
In case filebeat is configured with filestream input, kubernetes metadata are collected using the `add_kubernetes_metadata` processor. The <<add-kubernetes-metadata>> processor, by using matchers, tries to associate a log file detected by filestream input with an actual container. You can find more information about the available indexers and matchers, and some examples in <<kubernetes-indexers-and-matchers>>.
```